gaboon/commands/install.py

```diff
@@ -1,49 +1,301 @@
+            i for i in repo_install_path.parent.iterdir() if i not in existing
+        )
+        shutil.move(installed, repo_install_path)
+
+        # Update versions file
+        if versions_install_path.exists():
+            with open(versions_install_path, "rb") as f:
+                versions_data = tomllib.load(f)
+                versions_data[f"{org}/{repo}"] = version
+                tomli_w.dump(versions_data, f)
+        else:
+            with open(versions_install_path, "w", encoding="utf-8") as f:
+                toml_string = tomli_w.dumps({f"{org}/{repo}": version})
+                f.write(toml_string)
+        _write_dependencies(github_ids, DependencyType.GITHUB)
+
+
+def _get_latest_version(org: str, repo: str, headers: dict) -> str:
+    response = requests.get(
+        f"https://api.github.com/repos/{org}/{repo}/releases/latest", headers=headers
+    )
+    if response.status_code == 200:
+        return response.json()["tag_name"].lstrip("v")
+
+    response = requests.get(
+        f"https://api.github.com/repos/{org}/{repo}/tags?per_page=1", headers=headers
+    )
+    if response.status_code == 200:
+        data = response.json()
+        if data:
+            return data[0]["name"].lstrip("v")
+
+    raise ValueError(f"Unable to determine latest version for {org}/{repo}")
+
+
+def _stream_download(
+    download_url: str, target_path: str, headers: dict[str, str] = REQUEST_HEADERS
+) -> None:
+    response = requests.get(download_url, stream=True, headers=headers)
+
+    if response.status_code == 404:
+        raise ConnectionError(
+            f"404 error when attempting to download from {download_url} - "
+            "are you sure this is a valid mix? https://github.com/brownie-mix"
+        )
+    if response.status_code != 200:
+        raise ConnectionError(
+            f"Received status code {response.status_code} when attempting "
+            f"to download from {download_url}"
+        )
+
+    total_size = int(response.headers.get("content-length", 0))
+    progress_bar = tqdm(total=total_size, unit="iB", unit_scale=True)
+    content = bytes()
+
+    for data in response.iter_content(1024, decode_unicode=True):
+        progress_bar.update(len(data))
+        content += data
+    progress_bar.close()
+
+    with zipfile.ZipFile(BytesIO(content)) as zf:
+        zf.extractall(target_path)
+
+
+def _maybe_retrieve_github_auth() -> dict[str, str]:
+    """Returns appropriate github authorization headers.
+
+    Otherwise returns an empty dict if no auth token is present.
+    """
+    token = os.getenv("GITHUB_TOKEN")
+    if token:
+        auth = b64encode(token.encode()).decode()
+        return {"Authorization": f"Basic {auth}"}
+    return {}
+
+
+def _get_download_url_from_tag(org: str, repo: str, version: str, headers: dict) -> str:
+    response = requests.get(
+        f"https://api.github.com/repos/{org}/{repo}/tags?per_page=100", headers=headers
+    )
+    if response.status_code != 200:
```

Comment:
honestly, `response.raise_for_status()` works fine here

Reply:
fixed in 1b1b2b5